src/atldld/utils.py

```diff
@@ -95,60 +96,44 @@ def get_image(image_id, folder=None, expression=False, downsample=0):
     Raises
     ------
     ValueError
-        If request to download the image failed.
+        If the image has a wrong format (determined by the dtype).
 
     References
     ----------
     [1] `AllenSDK API: ImageDownloadApi <https://allensdk.readthedocs.io/
     en/latest/allensdk.api.queries.image_download_api.html#allensdk.api.
     queries.image_download_api.ImageDownloadApi>`_
-
     """
     folder = folder or CACHE_FOLDER
-    folder = str(
-        folder
-    )  # this should guarantee that also LocalPath works (pytest uses it)
-
-    # Check the folder exists
-    if not os.path.exists(folder):
-        os.makedirs(folder)
-
-    # Create full path
-    additional_specifier = "_expression" if expression else ""
-    additional_specifier += f"_{downsample}"
-    path = "{}{}{}.jpg".format(folder, image_id, additional_specifier)
-
-    # Check image exists
-    if os.path.exists(path):
-        img = plt.imread(path)
-
-        if not img.dtype == np.uint8:
-            raise ValueError("The dtype needs to be uint8")
-
-        return img
-
-    else:
-        image_url = (
-            f"https://api.brain-map.org/api/v2/section_image_download/{image_id}"
-        )
-        options = []
-
+    folder = pathlib.Path(folder)
+    folder.mkdir(exist_ok=True, parents=True)
+
+    # Construct the image file name and the full path
+    file_name = f"{image_id}-{downsample}"
+    if expression:
+        file_name += "-expression"
+    image_path = folder / (file_name + ".jpg")
+
+    # Download the image if not already in the cache
+    if not image_path.exists():
+        base_url = "https://api.brain-map.org/api/v2/section_image_download"
+        url = f"{base_url}/{image_id}?downsample={downsample}"
```

Comment:
I am just sharing my thoughts here! 

1. It is not introduced by this PR, but should we have a check that the downsampling make senses ? Moreover, we know that from a certain downsample value, the image shape is constant. Should we have a check/warning for the user ?
2. This current implementation could lead to one image being saved several times with different shapes. Do you think it could be worth thinking about this and trying to optimise in that sense ?

Reply:
1. I guess you are referring to the fact that downloaded image never seems to go below `100` in height/width, right? (No matter how big the `downsample` was). IMO we can definitely mention this in the docstring so that the user does not fall into this "trap". However, since the default value `downsample=0` leads to no downsampling I think we do not have to specifically write checks.

Reply:
Good point. I think if there was a fixed maximal value for `downsample` I would have said yes straight away. But like Honza said, it actually depends on the image size, which we don't know at this point. So a check would require at least one additional query to find out the image dimensions, which I'm not sure I like.

Reply:
By the way, @EmilieDel @jankrepl, now that we're in this block of code, do you guys know if there's any difference between `https://api.brain-map.org/api/v2/section_image_download` and `https://api.brain-map.org/api/v2/image_download`? The [docs about downloading images](http://help.brain-map.org/display/api/Downloading+an+Image) mention the latter. Where does that former that we use in our code come from?

Reply:
I think originally we used the `allensdk` to donwload images. Anyway, I think @EmilieDel might know the answer.

Reply:
Yes, the `url` was chosen when we removed `allensdk` dependency that was initially used to download images. If I remember correctly, I saw the `section_image_download` url on [this webpage at the cell 6](https://allensdk.readthedocs.io/en/latest/_static/examples/nb/image_download.html). To be honest, I don't really know the difference between the two. As it seems it is the`image_download` one mentioned in the documentation, I don't mind switching for it :)

Reply:
Ah, nice. I looked through the corresponding code, [they define the endpoings for the image download in the `Api` class](https://allensdk.readthedocs.io/en/latest/_modules/allensdk/api/api.html#Api.set_api_urls).

Looking through their code it seems that for section images they always use the [`v2/section_image_download` endpoint](https://allensdk.readthedocs.io/en/latest/_modules/allensdk/api/queries/image_download_api.html#ImageDownloadApi.download_section_image) and [only use `v2/image_download` as a generic fallback if no other endpoint was specified](https://allensdk.readthedocs.io/en/latest/_modules/allensdk/api/queries/image_download_api.html#ImageDownloadApi.download_image).

Anyway, I think `section_image_download` is better as it's more specific. So I suggest we keep it that way.